src/main/scala/contadamination/ContaminationModel.scala

```diff
@@ -0,0 +1,179 @@
+package contadamination
+
+import com.twitter.algebird.{BloomFilterMonoid, BloomFilter, BF}
+import contadamination.ContaminationModel.{ContaminationStats, FilterContext}
+import org.apache.spark.mllib.rdd.RDDFunctions
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.rdd.ADAMContext
+import org.bdgenomics.formats.avro.AlignmentRecord
+
+import scala.collection
+
+object ContaminationModel {
+
+  val defaultWinSize = 32
+  val defaultFpr = 0.1
+  val defaultSeed = 0
+
+  case class FilterContext(
+                name: String,     // Contig name
+                path: String,     // Reference path
+                numEntries: Int,  // Number of sliding windows
+                fpr: Double,      // False Positive Rate
+                bloomFilter: BF            // Bloom filter
+  )
+
+  case class ContaminationStats(
+                numReads: Int,    // Total number of reads checked
+                tn: Int,          // True Negatives (number of rejected reads)
+                rate: Double      // Contamination rate as the number of rejected reads over the total number of reads
+  )
+  
+  /**
+   * Build a ContaminationModel from a sequence of paths
+   * @param paths Sequence of FASTA files for the reference contigs
+   * @param winSize Sliding window size
+   * @param fpr False Positive Rate
+   * @param adamContext ADAM Context
+   * @return a ContaminationModel
+   */
+  def apply(paths: Seq[String], winSize: Int = defaultWinSize, fpr: Double = defaultFpr)(implicit adamContext: ADAMContext): ContaminationModel = {
+    paths.map(ContaminationModel(_, winSize, fpr)).reduce((x, y) => x ++ y)
+  }
+  
+  /**
+   * Build a ContaminationModel from a path
+   * @param path FASTA file with the reference contig
+   * @param winSize Sliding window size
+   * @param fpr False Positive Rate for the BloomFilter
+   * @param adamContext ADAM Context
+   * @return a ContaminationModel
+   */
+  def apply(path: String, winSize: Int = defaultWinSize, fpr: Double = defaultFpr)(implicit adamContext: ADAMContext): ContaminationModel = {
+    val fragments = adamContext.loadFasta(path, winSize).cache()
+    require(fragments.count() > 0, "At least one fragment is required")
+
+    // TODO check that there is only one contigName per file (right now we assume there is only one contig per file)
```

Comment:
Cool, I have some ideas in mind to fix this, but I didn't want to spend time until you confirmed me that this was really necessary. In fact this should be relatively easy as the model is ready to work with several contigs by design.